### ISSUE TITLE
Yet another false positive E0601 used-before-assignment, now for type hints

### ISSUE THREAD

**Issue by zEdS15B3GCwq** [PyCQA/pylint#8110]
### Issue
Code:
```Python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from numpy import ndarray

a: ndarray # E0601 here
b=ndarray()
```

Note that **only the type hint** was flagged by the error, the assignment was not.

### Related to
#7574 (may be fixed by #8071)
#7979 (fixed in #7980)
#7368 (fixed by #7360)
However, none of these seem to cover type hints.

### Command used

```shell
pylint --disable=missing-module-docstring .\pylint_test.py
```

### Pylint output

```shell
************* Module pylint_test
pylint_test.py:6:3: E0601: Using variable 'ndarray' before assignment (used-before-assignment)
```

### Expected behavior

No issues for the type hint.

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.3
Python 3.11.1 (tags/v3.11.1:a7a450f, Dec  6 2022, 19:58:39) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Windows 11 x64 21H2

**Comment by jacobtylerwalls** [comment]
Isn't this expected? This raises `NameError`:

```python
$ python3 a.py
Traceback (most recent call last):
  File "/Users/.../a.py", line 6, in <module>
    a: ndarray # E0601 here
NameError: name 'ndarray' is not defined
```

**Comment by zEdS15B3GCwq** [comment]
Eh, you're right. I'm sorry, should've tested that piece of code before posting. Following your response, I tried to write an example that was error-free but still produced E0601, but found that 2.15.10 was indeed working correctly and not reporting E0601.

Thank you for taking the time to reply and sorry for your trouble.

**Comment by nhairs** [comment]
So in case others end up here:

The reason that the snippet is throwing an error is because it uses [PEP 563 – Postponed Evaluation of Annotations](https://peps.python.org/pep-0563/) (3.7+) which requires `from __future__ import annotations`.

Applying this import will fix the runtime and pylint errors.
